File: platform/spigot-legacy/src/main/java/app/simplecloud/api/platform/spigot/legacy/PlayerConnectionListener.java
package app.simplecloud.api.platform.spigot.legacy;

import app.simplecloud.api.platform.shared.PlayerSynchronizer;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;
import org.bukkit.event.player.PlayerQuitEvent;

import java.util.concurrent.CompletableFuture;

public class PlayerConnectionListener implements Listener {

    private final PlayerSynchronizer playerSynchronizer;

    public PlayerConnectionListener(PlayerSynchronizer playerSynchronizer) {
        this.playerSynchronizer = playerSynchronizer;
    }

    @EventHandler
    public void onPlayerJoin(PlayerJoinEvent event) {
        CompletableFuture.runAsync(() -> playerSynchronizer.updatePlayerCount());
    }

    @EventHandler
    public void onPlayerQuit(PlayerQuitEvent event) {
        CompletableFuture.runAsync(() -> playerSynchronizer.updatePlayerCount());
    }
}

File: platform/spigot/src/main/kotlin/app/simplecloud/api/platform/spigot/PlayerConnectionListener.kt
package app.simplecloud.api.platform.spigot

import app.simplecloud.api.platform.shared.PlayerSynchronizer
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent
import org.bukkit.event.player.PlayerQuitEvent

class PlayerConnectionListener(
    private val playerSynchronizer: PlayerSynchronizer
) : Listener {

    @EventHandler
    fun onPlayerJoin(event: PlayerJoinEvent) {
        CoroutineScope(Dispatchers.Default).launch {
            playerSynchronizer.updatePlayerCount()
        }
    }

    @EventHandler
    fun onPlayerQuit(event: PlayerQuitEvent) {
        CoroutineScope(Dispatchers.Default).launch {
            playerSynchronizer.updatePlayerCount()
        }
    }

}